controllers/routemonitor/routemonitor_supplement.go

```diff
@@ -235,7 +236,24 @@ func (r *RouteMonitorReconciler) EnsureRouteURLExists(route routev1.Route, route
 	}
 
 	if currentRouteURL != "" && extractedRouteURL != currentRouteURL {
-		r.Log.V(3).Info("RouteURL mismatch: currentRouteURL and extractedRouteURL are not equal, taking extractedRouteURL as source of truth")
+		client := &http.Client{
+			CheckRedirect: func(req *http.Request, via []*http.Request) error {
+				return http.ErrUseLastResponse
+			},
+		}
+
+		resp, err := client.Head(extractedRouteURL)
+		if err != nil {
+			r.Log.V(3).Error(err, "Failed to make HEAD request to the URL")
+		}
+
+		if resp != nil && (resp.StatusCode == http.StatusMovedPermanently || resp.StatusCode == http.StatusFound || resp.StatusCode == http.StatusSeeOther || resp.StatusCode == http.StatusTemporaryRedirect || resp.StatusCode == http.StatusPermanentRedirect) {
+			r.Log.V(3).Info("RouteURL mismatch: redirection status code received, taking defaultRouteURL as source of truth")
+			extractedRouteURL = resp.Header.Get("Location")
+			defer resp.Body.Close()
```

Comment:
Should this defer be moved to outside the if/else block, so the response object is always closed? Or do we want to not close if we receive a bad response for some reason?